microraft/build.gradle.kts

```diff
@@ -0,0 +1,132 @@
+plugins {
+    `java-library`
+    `java-test-fixtures`
+    alias(libs.plugins.defaults)
+    alias(libs.plugins.metadata)
+    alias(libs.plugins.javadocLinks)
+    `maven-publish`
+    signing
+    alias(libs.plugins.mavenCentralPublishing)
+}
+
+group = "io.microraft"
+version = "0.9-SNAPSHOT"
+description = "Feature-complete implementation of the Raft consensus algorithm"
+
+metadata {
+    moduleName = "io.microraft"
+    readableName = "Microraft"
+    license {
+        apache2()
+    }
+    organization {
+        name = "MicroRaft"
+        url = "https://microraft.io"
+    }
+    developers {
+        register("metanet") {
+            fullName = "Ensar Basri Kahveci"
+            email = "ebkahveci@gmail.com"
+        }
+        register("mdogan") {
+            fullName = "Mehmet Dogan"
+            email = "mehmet@dogan.io"
+        }
+    }
+    github {
+        org = "MicroRaft"
+        pages()
+        issues()
+    }
+}
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(11)
+    }
+    withJavadocJar()
+    withSourcesJar()
+}
+
+tasks.withType<Jar>().configureEach {
+    manifest.attributes(
+        "Implementation-Title" to project.name,
+        "Implementation-Vendor" to metadata.organization.provider.flatMap { it.name },
+        "Implementation-Version" to provider { project.version.toString() },
+    )
+}
+
+tasks.javadoc {
+    isFailOnError = false
+    exclude("io/microraft/model/**")
+    exclude("**/impl/**")
+}
+
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    compileOnly(libs.findbugs)
+    implementation(libs.slf4j)
+}
+
+
+@Suppress("UnstableApiUsage") //
+testing {
+    suites {
+        withType(JvmTestSuite::class) {
+            useJUnit(libs.versions.junit)
+        }
+
+        val test by getting(JvmTestSuite::class) {
+            dependencies {
+                implementation(libs.mockito)
+                implementation(libs.assertj)
+                implementation(libs.log4j.core)
+                implementation(libs.log4j.api)
+                implementation(libs.log4j.slf4j.impl)
```

Comment:
these log4j dependencies can probably be runtimeOnly
log4j-api can then also be removed if it is not used during compile time